packages/contracts/contracts/Devcon6.sol

```diff
@@ -2,43 +2,56 @@
 
 pragma solidity 0.8.10;
 
+import "@openzeppelin/contracts/access/Ownable.sol";
+
 import "./Config.sol";
 import "./models/BidModel.sol";
+import "./models/StatusModel.sol";
+
+contract Devcon6 is Ownable, Config, BidModel, StatusModel {
+    uint256[] _auctionWinners;
+    uint256[] _raffleWinners;
+    uint256[] _raffleParticipants;
+    bool _notEnoughParticipantsForAuction;
 
-contract Devcon6 is Config, BidModel {
     mapping(address => Bid) _bids;
     // bidderID -> address
     mapping(uint256 => address) _bidders;
-    uint256 _nextBidderID = 0;
+    uint256 _nextBidderID = 1;
 
     constructor(
-        uint256 startTime,
-        uint256 endTime,
+        address initialOwner,
+        uint256 biddingStartTime,
+        uint256 biddingEndTime,
+        uint256 claimingEndTime,
         uint256 auctionWinnersCount,
         uint256 raffleWinnersCount,
         uint256 reservePrice,
         uint256 minBidIncrement
     )
         Config(
-            startTime,
-            endTime,
+            biddingStartTime,
+            biddingEndTime,
+            claimingEndTime,
             auctionWinnersCount,
             raffleWinnersCount,
             reservePrice,
             minBidIncrement
         )
-    {}
+        Ownable()
+    {
+        if (initialOwner != msg.sender) {
+            Ownable.transferOwnership(initialOwner);
+        }
+    }
 
     event NewBid(address bidder, uint256 bidID, uint256 bidAmount);
 
     function bid() external payable {
+        Status status = getStatus();
         require(
-            block.timestamp >= _startTime,
-            "Devcon6: bidding is not open yet"
-        );
-        require(
-            block.timestamp < _endTime,
-            "Devcon6: bidding is already closed"
+            status == Status.BIDDING_OPEN,
+            "Devcon6: is not in bidding open state"
         );
```

Comment:
Let's sprinkle some Solidity syntax sugar here and introduce a modifier `onlyInState(State.X)` modifier.